### ISSUE TITLE
[apex] ApexAssertionsShouldIncludeMessage and ApexUnitTestClassShouldHaveAsserts: support new Assert class (introduced with Apex v56.0)

### ISSUE THREAD

**Issue by tprouvot** [pmd/pmd#4096]
**Affects PMD Version:** 
all

**Rules:**
* [ApexAssertionsShouldIncludeMessage](https://pmd.github.io/latest/pmd_rules_apex_bestpractices.html#apexassertionsshouldincludemessage)
* [ApexUnitTestClassShouldHaveAsserts](https://pmd.github.io/latest/pmd_rules_apex_bestpractices.html#apexunittestclassshouldhaveasserts)

**Description:**
With the Winter 23' release, a new class can be used to perform assertion in test class.
We have to add the new methods to existing rule to avoid false positive when this class will be used.

[Apex class Assert](https://developer.salesforce.com/docs/atlas.en-us.240.0.apexref.meta/apexref/apex_class_System_Assert.htm#apex_System_Assert_methods)
**Code Sample demonstrating the issue:**

```java
@isTest
public with sharing class Test {

	@isTest
	public static void testAssert(){
		String sub = 'abcde'.substring(2);
		Assert.areEqual('cde', sub, 'Expected characters after first two');
	}
}

```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.
![image](https://user-images.githubusercontent.com/35368290/185136707-b81b0674-416f-4558-b055-bb66470b74a7.png)

**Running PMD through:** *[VS Code]*

**Comment by tprouvot** [comment]
Hi guys @adangel @rsoesemann @SCWells72,
I would be more than happy to contribute to the Apex PMD plugin, this is [my first PR ](https://github.com/pmd/pmd/pull/4097)on this repo so feel free to add comments or suggestions !
